ipam/allocator.go

```diff
@@ -122,19 +112,68 @@ func (a *Allocator) updateBitMasks(aSpace *addrSpace) error {
 func (a *Allocator) checkConsistency(as string) {
 	// Retrieve this address space's configuration and bitmasks from the datastore
 	a.refresh(as)
+	a.Lock()
 	aSpace, ok := a.addrSpaces[as]
+	a.Unlock()
 	if !ok {
 		return
 	}
 	a.updateBitMasks(aSpace)
+	aSpace.Lock()
```

Comment:
I still would like to avoid it. We cannot look at today's code and break a practice that we are following across other parts of the code. It will cause unncessary headaches in the long run, because of this particular assumption that it will not cause a deadlock with the current implementation.